go/filemanager/pfsmodules/remotefile.go

```diff
@@ -0,0 +1,157 @@
+package pfsmodules
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"mime/multipart"
+	"os"
+
+	"github.com/PaddlePaddle/cloud/go/utils/restclient"
+	log "github.com/golang/glog"
+)
+
+// RFileHandle is a remote file's handle.
+type RFileHandle struct {
+	Path string
+	Flag int
+	Size int64
+}
+
+const (
+	// ReadOnly means read only.
+	ReadOnly = os.O_RDONLY
+	// WriteOnly means write only.
+	WriteOnly = os.O_WRONLY
+	// ReadAndWrite means read and write.
+	ReadAndWrite = os.O_RDWR
+)
+
+// Open file to read ,write or read-write.
+// if flag == WriteOnly or flag == ReadAndWrite, this function will
+// attempt to create a sized file on remote if it does't exist.
+func (f *RFileHandle) Open(path string, flag int, size int64) error {
```

Comment:
It's a little confusing that this function is called `Open` but it does not open a file.